haskell/test/Model/BaseDAO/Proposal.hs

```diff
@@ -0,0 +1,420 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Model.BaseDAO.Proposal
+  ( applyFreeze
+  , applyUnfreeze
+  , applyVote
+  , applyPropose
+  , applyFlush
+  , applyDropProposal
+  , applyUpdateDelegate
+  )where
+
+import Universum
+
+import Control.Monad.Except (throwError)
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import GHC.Natural
+
+import Lorentz hiding (not, cast, get, take)
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import qualified Michelson.Typed as T
+
+import Ligo.BaseDAO.Types
+import Model.BaseDAO.Permit
+import Model.BaseDAO.Proposal.FreezeHistory
+import Model.BaseDAO.Proposal.QuorumThreshold
+import Model.BaseDAO.Token
+import Model.BaseDAO.Types
+import Test.Ligo.BaseDAO.Common
+
+checkIfProposalExist
```

Comment:
looks like this will need an update after #291 

please don't make it exactly the same, I suggest to have the two functions defined as one, aka the opposite of [this comment](https://github.com/tqtezos/baseDAO/pull/291#discussion_r661504171)

Reply:
I've updated this.